libenv/unix_iface.c

```diff
@@ -752,4 +755,370 @@ static int aix_get_mac_addr(const char *device_name, uint8_t mac[6])
 }
 #endif /* _AIX */
 
+/*******************************************************************/
+
+// returns NULL on any failure
+JsonElement* StringCaptureData(const char* regex, const char* data)
+{
+    Seq *s = StringMatchCaptures(regex, data, true);
+
+    if (!s || SeqLength(s) == 0)
+    {
+        SeqDestroy(s);
+        return NULL;
+    }
+
+    JsonElement *json = JsonObjectCreate(SeqLength(s)/2);
+
+    for (int i = 0; i < SeqLength(s); i+=2)
+    {
+        Buffer *key = SeqAt(s, i);
+        Buffer *value = SeqAt(s, i+1);
```

Comment:
In case you get an odd number of elements. Probably you won't, but better safe than sorry.